turbopack/crates/turbopack-swc-utils/src/emitter.rs

```diff
@@ -12,26 +14,75 @@ use turbopack_core::{
     source::Source,
 };
 
-#[derive(Clone)]
+pub struct IssueCollector {
+    inner: Arc<Mutex<IssueCollectorInner>>,
+}
+
+impl IssueCollector {
+    pub async fn emit(self) -> Result<()> {
+        let issues = {
+            let mut inner = self.inner.lock();
+            if inner.emitted {
+                return Ok(());
+            }
+            inner.emitted = true;
+            take(&mut inner.emitted_issues)
+        };
+
+        for issue in issues {
+            issue.to_resolved().await?.emit();
+        }
```

Comment:
This could happen in parallel? similar to `try_join()` (except there's no return result in this case)